src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/notifications/CaseNotesNotificationsService.kt

```diff
@@ -22,15 +24,24 @@ class CaseNotesNotificationsService(
   private val hmppsAuthService: HMPPSAuthService,
   private val userTypeChecker: UserTypeChecker,
   private val communityAPIOffenderService: CommunityAPIOffenderService,
+  private val caseNoteRepository: CaseNoteRepository,
 ) : ApplicationListener<CaseNoteEvent>, NotifyService {
   companion object : KLogging()
 
   @AsyncEventExceptionHandling
   override fun onApplicationEvent(event: CaseNoteEvent) {
     when (event.type) {
       CaseNoteEventType.SENT -> {
-        emailAssignedCaseWorker(event.caseNote)
-        emailResponsibleProbationPractitioner(event.caseNote)
+        // get case note from repository because the linking objects (referral.currentAssignee) on the caseNote from the event
+        // may sometimes not be lazy loaded because the session may already have been closed by this point
+       caseNoteRepository
+         .findById(event.caseNote.id).ifPresentOrElse({
+           emailAssignedCaseWorker(it)
+           emailResponsibleProbationPractitioner(it)
+         },{
+           // should never happen
+          logger.error("Unable to retrieve case note for id ${event.caseNote.id}")
```

Comment:
As this is an unrecoverable failure, could we use `orElseThrow` here instead? That seems semantically to be closer to the intended behaviour, plus makes it implicit that it should not happen

Reply:
Yes, it would be nicer that way. I've changed to an exception.